src/pages/ArticleList.jsx

```diff
@@ -16,21 +18,78 @@ const ArticleList = () => {
       query.set('topic', slug);
     }
 
-    getAllArticles(query).then(({ articles }) => {
+    const combinedQuery = new URLSearchParams({
+      ...Object.fromEntries(query),
+      ...Object.fromEntries(searchParams),
+    });
+
+    getAllArticles(combinedQuery).then(({ articles }) => {
       setArticles(articles);
     }).catch(() => {
       setHasLoadingError(true);
     }).finally(() => {
       setIsLoading(false);
-    });;
-  }, [slug]);
+    });
+
+    scrollTo(0, 0);
```

Comment:
Is this necessary? 👀

Reply:
At times yes because I have a link in the article card to navigate to the relevant topic, and because I am reusing the `ArticleList` component, the original scroll position was being maintained.

You were referring to the scrollTo() right?  😃